Networking/BLIP/BLIPConnection.cc

```diff
@@ -283,7 +281,9 @@ namespace litecore { namespace blip {
         void requeue(MessageOut *msg, bool andWrite =false) {
             DebugAssert(!_outbox.contains(msg));
             auto i = _outbox.end();
-            if (msg->urgent() && _outbox.size() > 1) {
+            // Only allow unnumbered msg to be inserted.
+            // Numbered msg must be appended
+            if (msg->_number == 0 && msg->urgent() && _outbox.size() > 1) {
```

Comment:
This is incorrect. Please try without it; the other change below should fix the bug.

Reply:
If a numbered msg is inserted, say, m1 m3 m2, then at the time it is pulled out from the outbox, we will have out-of-order and triggering "too high" exception.

Reply:
Only the first frame of a message has to have the consecutive numbering.